src/modules/acbs/acbs-facility-covenant.service.replace-covenant.test.ts

```diff
@@ -0,0 +1,183 @@
+import { HttpService } from '@nestjs/axios';
+import { PROPERTIES } from '@ukef/constants';
+import { CreateFacilityCovenantGenerator } from '@ukef-test/support/generator/create-facility-covenant-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+
+import { DateStringTransformations } from '../date/date-string.transformations';
+import { AcbsFacilityCovenantService } from './acbs-facility-covenant.service';
+import { AcbsBadRequestException } from './exception/acbs-bad-request.exception';
+import { AcbsResourceNotFoundException } from './exception/acbs-resource-not-found.exception';
+import { AcbsUnexpectedException } from './exception/acbs-unexpected.exception';
+
+describe('AcbsFacilityCovenantService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+  const authToken = valueGenerator.string();
+  const baseUrl = valueGenerator.httpsUrl();
+  const facilityIdentifier = valueGenerator.ukefId();
+  const facilityTypeCode = valueGenerator.stringOfNumericCharacters();
+  const limitKeyValue = valueGenerator.string();
+  const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
```

Comment:
not a showstopper - we can deconstruct this

Reply:
Changed